pkg/rdt/log.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2019 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rdt
+
+import (
+	stdlog "log"
+	"os"
+)
+
+type logger struct {
+	*stdlog.Logger
+}
+
+func NewLogger(prefix string) *logger {
+	return &logger{Logger: stdlog.New(os.Stdout, prefix, 0)}
+}
```

Comment:
It would allow better logging control/override for users of this package if you would
  - define and use an `interface Logger{}` here, and
  - either add a `control.SetLogger(l Logger)` to `rdt.go`, or
  - preferably change `Initialize()` to `Initialize(l Logger)` with an optional Logger to use

IOW, 
  - define an interface you expect a logger to implement
  - have a default internal implementation
  - let users plug in their logger of preference

You almost have it already. You would just need to spell out the interface and change Initialize() to create/use the fallback Logger if nil is passed in.

That's what for instance `grpc` does. Although it is possible to bridge logging without such a setup, there virtually is no control over anything because it's done by plugging in an alternative `io.Writer` to `stdlog` (like here https://github.com/klihub/cri-resource-manager/blob/devel/klogger/pkg/log/stdlog-logger.go), and there is no hope for improving that before golang 2.

Reply:
Mm, good and justified comment 👍  And even more so, as you said, it's almost done. I'll change the implementation, better to start with a more reasonable implementation 😄

Reply:
Implementation now changed